sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -1520,7 +1520,13 @@ private RxDocumentServiceRequest addBatchHeaders(RxDocumentServiceRequest reques
         return request;
     }
 
-    private Mono<RxDocumentServiceRequest> populateHeaders(RxDocumentServiceRequest request, RequestVerb httpMethod) {
+    /**
+     * NOTE: Caller needs to consume it by subscribing to this Mono in order for the request to populate headers
+     * @param request request to populate headers to
+     * @param httpMethod http method
+     * @return Mono, which on subscription will populate the headers in the request passed in the argument.
+     */
+    private Mono<RxDocumentServiceRequest> populateHeadersAsync(RxDocumentServiceRequest request, RequestVerb httpMethod) {
```

Comment:
is it a convention to add Async as suffix?

Reply:
populateHeaders has a bunch of non-Reactor code that executes for regular auth, but for AAD we need to invoke the underneath Mono. The bug we found was caller for the patch did not subscribe to the Mono part and somehow all the regular tests still passed. To avoid similar issues when we add other APIs perhaps the name change is enough of an indicator to let the developer know that they should subscribe to this internal method :-)

Reply:
It is not, you are right. But since someone missed subscribing to this, and this is internal, so added it to make sure no one misses it in the future.